Assembler.Core/Entities/BaseMessage.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace Assembler.Core.Entities
+{
+    public abstract class BaseMessage
+    {
+        public Guid Guid { get; }
+
+        protected BaseMessage(Guid guid)
```

Comment:
Why letting someone append guid and not create a new one by using the other CTor?

Reply:
Maybe you want to keep the same Guid you already have and just wrap it in this class